src/entrypoints/my-index.ts

```diff
@@ -7,99 +7,69 @@ import {
   internalProperty,
 } from "lit-element";
 import "../components/my-url-input";
-import { svgPencil } from "../components/svg-pencil";
-import { DEFAULT_HASS_URL, HASS_URL } from "../const";
-import { DiscoveryInfo, getHassInfo } from "../util/get_hass_info";
+import "../components/my-instance-info";
+import { LoadingState, subscribeHassInfo } from "../util/get_hass_info";
 import { extractSearchParamsObject } from "../util/search-params";
 
 @customElement("my-index")
 class MyIndex extends LitElement {
-  @internalProperty() private _url?: string | null = localStorage.getItem(
-    HASS_URL
-  );
-
   @internalProperty() private _updatingUrl =
     extractSearchParamsObject().change === "1";
 
-  @internalProperty() private _instanceInfo?: DiscoveryInfo | false;
+  @internalProperty() private _instanceInfo!: LoadingState;
 
   createRenderRoot() {
     return this;
   }
 
-  protected render(): TemplateResult {
-    const url = this._url || DEFAULT_HASS_URL;
+  public connectedCallback() {
+    super.connectedCallback();
+    this._updateInstanceInfo();
+  }
+
+  protected shouldUpdate() {
+    return this._instanceInfo !== undefined;
+  }
 
-    if (!this._updatingUrl) {
+  protected render(): TemplateResult {
+    if (this._updatingUrl) {
       return html`
-        <div class="instance-header">HOME ASSISTANT INSTANCE</div>
-        <div class="instance">
-          <div>
-            <div>
-              <a href=${url} rel="noreferrer noopener">${url}</a>
-            </div>
-            <div>
-              ${this._instanceInfo === undefined
-                ? html`&nbsp;`
-                : this._instanceInfo === false
-                ? html`<span class='error'>Failed to fetch info (<a href="faq.html#cannot-connect">troubleshoot</a>)</span>`
-                : `Version core-${this._instanceInfo.version}`
-              }
-            </div>
-            <div>
-              ${this._instanceInfo === undefined || this._instanceInfo === false
-                ? html`&nbsp;`
-                : `Installation type ${this._instanceInfo.installation_type}`
-              }
-            </div>
-          </div>
-          <button class="empty" @click=${this._handleEdit}>${svgPencil}</button>
+        <div class="card-content">
+          <p>
+            <my-url-input
+              .value=${this._instanceInfo!.url}
```

Comment:
Either this or correct the type of `_instanceInfo`
```suggestion
              .value=${this._instanceInfo.url}
```